overlord/ifacestate/ifacemgr.go

```diff
@@ -22,36 +22,91 @@
 package ifacestate
 
 import (
+	"github.com/ubuntu-core/snappy/interfaces"
+	"github.com/ubuntu-core/snappy/interfaces/builtin"
 	"github.com/ubuntu-core/snappy/overlord/state"
 )
 
 // InterfaceManager is responsible for the maintenance of interfaces in
 // the system state.  It maintains interface connections, and also observes
 // installed snaps to track the current set of available plugs and slots.
-type InterfaceManager struct{}
+type InterfaceManager struct {
+	// state holds a reference to persistent system state. Interface engine
+	// uses the state to store list of "intents-to-connect" that were made by
+	// the operator. Those intents are made effective during Ensure().
+	state *state.State
+	// repo contains all the interfaces, plugs and slots declared by snaps (or
```

Comment:
Would be nice to have a space after each field definition, and before the next block of text. As it is it's a continuum that doesn't highlight the fact the block of text is about the thing below it.